File: Resources/CourseraAlgorithmicToolbox/week3_greedy_algorithms/6_maximum_number_of_prizes/different_summands.py
# Uses python3
import sys

def optimal_summands(n):
    summands = []
    #write your code here
    return summands

if __name__ == '__main__':
    input = sys.stdin.read()
    n = int(input)
    summands = optimal_summands(n)
    print(len(summands))
    for x in summands:
        print(x, end=' ')

File: Resources/CourseraAlgorithmicToolbox/week3_greedy_algorithms/6_maximum_number_of_prizes/different_summands.hs
-- by Kirill Elagin

optimal_summands :: Int -> [Int]
optimal_summands n = [n] -- write your code here

main :: IO ()
main = do
  n  <- getLine
  let summands = optimal_summands (read n)
  print $ length summands
  putStrLn $ unwords (map show summands)
